stacks-signer/src/ping/mod.rs

```diff
@@ -0,0 +1,148 @@
+use std::{fmt, fmt::Debug};
+
+use rand_core::{OsRng, RngCore};
+use serde_derive::{Deserialize, Serialize};
+mod periodic_pinger;
+use crate::{
+    stacks_client::{PING_SLOT, SLOTS_PER_USER},
+    utils::ToSlotId,
+};
+pub use periodic_pinger::{PeriodicPinger, ThreadStopHandle};
+
+impl<'a> ToSlotId for &'a Packet {
+    fn slot_id(&self, signer_id: u32) -> u32 {
+        SLOTS_PER_USER * signer_id + PING_SLOT
+    }
+}
+
+/// Is an incoming slot update a ping::Packet?
+/// Use it to filter out other packets.
+pub fn is_ping_slot(slot_id: u32) -> bool {
+    let Some(v)=slot_id.checked_sub(PING_SLOT) else{
+        return false
+    };
+
+    v % SLOTS_PER_USER == 0
+}
+
+/// What is written to the ping slot.
+#[derive(Serialize, Deserialize, Clone)]
+pub enum Packet {
+    /// Outgoing
+    Ping(Ping),
+    /// Incoming
+    Pong(Pong),
+}
+
+#[derive(Serialize, PartialEq, Clone, Deserialize)]
+/// A ping in a slot means someone has requested you to push a Pong into your slot.
+pub struct Ping {
+    id: u64,
+    payload: Vec<u8>,
+}
+#[derive(Serialize, Deserialize, Clone)]
+/// A pong in a slot means someone has responded to an RTT request.
+pub struct Pong {
+    id: u64,
+    payload: Vec<u8>,
+}
+
+impl From<Pong> for Packet {
+    fn from(value: Pong) -> Self {
+        Self::Pong(value)
+    }
+}
+
+impl From<Ping> for Packet {
+    fn from(value: Ping) -> Self {
+        Self::Ping(value)
+    }
+}
+
+impl Ping {
+    /// Uniquely identify the RTT request
+    pub fn new(payload_size: usize) -> Self {
+        let mut payload = Vec::with_capacity(payload_size);
+        OsRng.fill_bytes(payload.as_mut_slice());
+        Ping {
+            id: OsRng.next_u64(),
+            payload,
+        }
+    }
+
+    /// Pong receives its fields from a ping.
+    pub fn pong(self) -> Pong {
+        Pong {
+            id: self.id,
+            payload: self.payload,
+        }
+    }
+
+    /// getter
+    pub fn id(&self) -> u64 {
+        self.id
+    }
+}
+
+impl Pong {
+    /// getter
+    pub fn id(&self) -> u64 {
+        self.id
+    }
+}
+
+macro_rules! debug_skip_payload_field {
+    ($($names:ident),*) => {
+        $(
+        impl Debug for $names {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                f.debug_struct(stringify!($names))
+                    .field("id", &self.id)
+                    .finish()
+            }
+        }
+        )*
+    };
+}
+
+debug_skip_payload_field!(Ping, Pong);
```

Comment:
This is overkill. Can you just implement `Debug` for `Ping` and `Pong` directly?